src/lib.rs

```diff
@@ -1561,6 +1581,21 @@ impl<R: Read + Seek> MatroskaFile<R> {
                     return Err(err);
                 }
             }
+
+            // Manage the end of the loop. Return true if :
+            // - a SimpleBlock found and read
+            // - a BlockGroup found and read to the end. This happens when it contains one Block (frame) + BlockDuration.
+            // - a BlockGroup found with laced frames decoded and pushed in queued_frames.
+            //   the first frame should have been popped in `frame` and other frames waiting in `queued_frames`
+            if block_end.is_some_and(|val| {
+                val == self
+                    .file
+                    .stream_position()
+                    .expect("stream_position shouldn't return error, at least for opened file")
```

Comment:
This is an io error, which we expose to the user. So you can just return the `DemuxError::IoError` here.